controller/lib/core/blower_fsm.cpp

```diff
@@ -235,9 +234,9 @@ BlowerSystemState BlowerFsm::DesiredState(Time now, const VentParams &params,
       break;
     }
   }
-
-  BlowerSystemState s =
-      std::visit([&](auto &fsm) { return fsm.DesiredState(); }, fsm_);
-  s.is_new_breath = is_new_breath;
+  if (switching_on) {
+    s = std::visit([&](auto &fsm) { return fsm.DesiredState(now, inputs); },
+                   fsm_);
+  }
```

Comment:
I think this is fine; the comment is very clear.

It seems like it would be even simpler not to switch the vent on immediately.  We could wait one cycle, that wouldn't be a big deal.  The communication latency is already 30ms or so, and the fan takes maybe a second to spin up, so adding an additional 10ms to the "on" command would be fine.

I think that would clean this code up a bit, but it would make the tests a little uglier.  So up to you.

Reply:
Choosing to keep this special case here since it's fairly clear-cut